Add Journey3Dg demo

doogle opened the pull request:
To reduce the source code size, all comments (except some header comments) have been stripped, and various other tidy-ups have been performed.  
This reduced the source size from ~100KiB to ~80KiB. If that's still a bit too much I can try to reduce it further. Another alternative would be to provide the .mpy files for the arcade, which would be ~57KiB instead. Please let me know if that would be preferred.

A link to the full source repo is provided in the description and in the main .py file.

jmtinycircuits commented:
I think the size is fine. As long as no out of memory error occurs when loading the game it should be good.

The video looks a bit messed up in the Windows video player, but OK in Chrome, so we'll just see how that works out in the Arcade.

Looks likes the emulator doesn't like to play this. I get this error:
```
>>> __import__('/Games/Journey3Dg/Journey3Dg')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Games/Journey3Dg/Journey3Dg.py", line 148, in <module>
  File "<string>", line 2, in <module>
  File "Games/Journey3Dg/Journey3Dg_main.py", line 37, in <module>
  File "/Games/Journey3Dg/ssd1306grey.py", line 37, in <module>
TypeError: unsupported types for __lt__: 'NoneType', 'int'
>>>
```

I'm fine with it not being playable in the emulator, you'll just need to add a note to the description saying that. Otherwise, it works fine on hardware.

doogle commented:
Ah, good point about the emulator. I didn't think it'd crash like that, but either it won't work in the emulator due to the greyscale stuff. I'll add a note in the description, and I'll put in a runtime check as well in case the user tries anyway.

That's weird about the video. Again, because of the greyscale I had to do things a bit differently, so I dumped .png images from a slightly modified version running under CPython and then used `ffmpeg` to generate the webm. I'll look into why it isn't working in the Windows video player and see if I can fix that. I have a Windows machine I can test it on here before I update this branch.

I think you were planning to anyway, but please hold off the merge for now and I'll push some changes soon.  
Would you prefer another commit on top, or are you ok if I amend the existing commit and force push? (I'd prefer the latter to keep the commit history clean).

jmtinycircuits commented:
I think that sounds fine for the emulator.

You don't need to look into the video. If it works in the browser then we're fine.

Go ahead and do what works for you for adding/editing commits.

doogle commented:
I've updated the description and added the runtime check as discussed (and made some other minor tweaks).
I've left the video file as-is.

From my side this should be ready to merge, but please let me know if there are any further changes needed.

jmtinycircuits commented:
Alright! Looks good.
